index/scorch/merge.go

```diff
@@ -265,7 +267,9 @@ func (s *Scorch) planMergeAtSnapshot(ourSnapshot *IndexSnapshot,
 			}
 		}
 	}
-
+	// once the newly merged segment introduction is done,
+	// its safe to unflip the removal ineligibility.
+	s.unmarkIneligibleForRemoval(filename)
```

Comment:
this seems to be the key change.

One thought, though, is sometimes filename might be ""?

And, another thought is there's a for-loop before this that ranges over several filenames, but this is unmarking a single filename?

Also, handing the segment off to the introducer channel might not be enough?  E.g., it's in the channel, but the introducer hasn't actually gotten around to doing its work yet?

Reply:
yikes... very true.. I went blind with this change..

Reply:
Once we receive the notifications back from the introducer, the introductions are actually done?, But need to take care of all the files.